docs: example pruning and nits (#3374)
fubuloubu/web3.py: docs/examples.rst

    175

> .. _ERC-20: https://github.com/ethereum/EIPs/blob/7f4f0377730f5fc266824084188cc17cf246932e/EIPS/eip-20.md

Reviewer: Separation of EIPs and ERCs

```suggestion
.. _ERC-20: https://github.com/ethereum/ERCs/blob/master/ERCS/erc-20.md
```

eth_getLogs limitations
~~~~~~~~~~~~~~~~~~~~~~~

> Ethereum JSON-RPC API servers, like Geth, do not provide easy to paginate over events, only over blocks. There's no request that can find the first block with an event or how many events occur within a range of blocks. The only feedback the JSON-RPC service will give you is whether the ``eth_getLogs`` call failed.

Reviewer 2: ```suggestion
Ethereum JSON-RPC API servers, like Geth, do not provide an easy way to paginate over events, only over blocks. There's no request that can find the first block with an event or how many events occur within a range of blocks. The only feedback the JSON-RPC service will give you is whether the ``eth_getLogs`` call failed.
```

In this example script, we provide two kinds of heuristics to deal with this issue. The script scans events in a chunk of blocks (start block number - end block number). Then it uses two methods to find how many events there are likely to be in a block window:

* Dynamically set the block range window size, while never exceeding a threshold (e.g., 10,000 blocks).

> * In the case ``eth_getLogs`` JSON-PRC call gives a timeout error, decrease the end block number and try again with a smaller block range window.

Reviewer 2: ```suggestion
* In the case of ``eth_getLogs``, the JSON-RPC call gives a timeout error, decrease the end block number and tries again with a smaller block range window.
```
